In [1]:
import os, re
import pandas as pd
os.chdir(r'C:\Users\kke86\OneDrive\바탕 화면\data_python\data_python\Basic')
df = pd.read_csv('survey.csv',encoding = 'cp949')

In [2]:
df.head()

sex  income  English  jobSatisfaction  stress
0   m    3000      500                5       5
1   f    4000      600                4       4
2   f    5000      700                3       2
3   m    6000      800                2       2
4   m    4000      700                2       5

In [5]:
df.mean()      # 평균 

C:\Users\kke86\AppData\Local\Temp\ipykernel_14332\504831637.py:1: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  df.mean()      # 평균


income             4304.217391
English             608.695652
jobSatisfaction       3.304348
stress                3.347826
dtype: float64

In [6]:
df.income.mean()     # 수입의 평균

4304.217391304348

In [7]:
df.income.sum()     # 수입의 합계

98997

In [8]:
df.income.median()     # 수입의 중앙값

4999.0

In [9]:
df.describe()     # 기초 통계량

income     English  jobSatisfaction     stress
count    23.000000   23.000000        23.000000  23.000000
mean   4304.217391  608.695652         3.304348   3.347826
std    1019.478341   99.603959         1.258960   1.433644
min    3000.000000  500.000000         1.000000   1.000000
25%    3000.000000  500.000000         2.500000   2.000000
50%    4999.000000  600.000000         3.000000   4.000000
75%    5000.000000  700.000000         4.000000   5.000000
max    6000.000000  800.000000         5.000000   5.000000

In [10]:
df.income.describe()     # 수입의 기초 통계량

count      23.000000
mean     4304.217391
std      1019.478341
min      3000.000000
25%      3000.000000
50%      4999.000000
75%      5000.000000
max      6000.000000
Name: income, dtype: float64

빈도 분석 : value_counts()

In [11]:
df.sex.value_counts()

m    14
f     9
Name: sex, dtype: int64

두 집단 평균 구하기

In [12]:
df.describe()

income     English  jobSatisfaction     stress
count    23.000000   23.000000        23.000000  23.000000
mean   4304.217391  608.695652         3.304348   3.347826
std    1019.478341   99.603959         1.258960   1.433644
min    3000.000000  500.000000         1.000000   1.000000
25%    3000.000000  500.000000         2.500000   2.000000
50%    4999.000000  600.000000         3.000000   4.000000
75%    5000.000000  700.000000         4.000000   5.000000
max    6000.000000  800.000000         5.000000   5.000000

In [13]:
df.groupby(df.sex).mean()

income     English  jobSatisfaction    stress
sex                                                    
f    4333.111111  633.333333         3.666667  3.111111
m    4285.642857  592.857143         3.071429  3.500000

In [14]:
from scipy import stats

### < t-검정 >

In [17]:
male = df.income[df.sex == 'm']      # 남성의 수입
female = df.income[df.sex == 'f']    # 여성의 수입

stats.ttest_ind(male, female)      # t검정

Ttest_indResult(statistic=-0.10650308143428425, pvalue=0.9161940781163369)

pvalue(유의확률)이 작을수록 유의한 차이가 있다. 위에 결과는 pvalue가 0.916으로 1에 가까울 정도로 높다. 

즉 남성과 여성의 수입평균을 비교한 t검정의 결과에 유의한 차이가 있다고 보기는 어렵다.

In [18]:
ttsse_result = stats.ttest_ind(male, female) 
print(ttsse_result)

Ttest_indResult(statistic=-0.10650308143428425, pvalue=0.9161940781163369)


In [20]:
print(ttsse_result[0])    # statics 값이 들어감
print(ttsse_result[1])    # pvalue 값이 들어감

-0.10650308143428425
0.9161940781163369


In [21]:
if ttsse_result[1] > .05:
    print('p-value는 %f로 95%% 수준에서 유의하지 않음' % ttsse_result[1])
else:
     print('p-value는 %f로 95%% 수준에서 유의함' % ttsse_result[1])


p-value는 0.916194로 95% 수준에서 유의하지 않음


### < 상관관계 분석 >
- 스피어만의 상관계수 : 변수가 순위 척도 일 때 사용
- 피어슨의 상관계수 : 연속형자의 상관관계를 다룬다.

In [22]:
corr = df.corr()
corr

income   English  jobSatisfaction    stress
income           1.000000  0.599452        -0.040108 -0.137920
English          0.599452  1.000000        -0.312051  0.073351
jobSatisfaction -0.040108 -0.312051         1.000000  0.165338
stress          -0.137920  0.073351         0.165338  1.000000

In [23]:
df.corr(method= 'spearman')


income   English  jobSatisfaction    stress
income           1.000000  0.543705        -0.100683 -0.170584
English          0.543705  1.000000        -0.309747  0.068223
jobSatisfaction -0.100683 -0.309747         1.000000  0.154982
stress          -0.170584  0.068223         0.154982  1.000000

In [26]:
df.income.corr(df.stress)     # 수입과 스트레스의 상관관계

-0.13791959796123449

### < 단순 회귀 분석 >

In [33]:
import statsmodels.formula.api as smf

model = smf.ols(formula = 'jobSatisfaction~English', data =df)
result = model.fit()

print(result.summary())


                            OLS Regression Results                            
Dep. Variable:        jobSatisfaction   R-squared:                       0.097
Model:                            OLS   Adj. R-squared:                  0.054
Method:                 Least Squares   F-statistic:                     2.266
Date:                Tue, 27 Jun 2023   Prob (F-statistic):              0.147
Time:                        15:56:10   Log-Likelihood:                -36.243
No. Observations:                  23   AIC:                             76.49
Df Residuals:                      21   BIC:                             78.76
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      5.7052      1.615      3.532      0.0

[ 결정계수(R2) ]  
- 해당 모델이 관측치에 얼마나 잘 맞는지 알려주는 지표
- 0~1 의 값을 갖는다. 모델이 완전히 관측치에 들어맞을수록 1에 가까워 진다.
- 보통 03.~0.7 사이에 값이 분포한다. 1에 너무 가까우면 연구 가치가 없는 것으로 해석될 여지가 있음.
- 위 모델에서는 R-squared의 값이 0.097로 너무 낮음

[ 유의확률 Prob(F-statistic) ]

[ 계숫값 지표 coef(cofficient) ]
- 위 모델은 English으 coef값은 -0.0039

In [34]:
model2 = smf.ols(formula = 'jobSatisfaction~English + stress + income', data =df)
result = model.fit()

print(result.summary())

                            OLS Regression Results                            
Dep. Variable:        jobSatisfaction   R-squared:                       0.097
Model:                            OLS   Adj. R-squared:                  0.054
Method:                 Least Squares   F-statistic:                     2.266
Date:                Tue, 27 Jun 2023   Prob (F-statistic):              0.147
Time:                        16:36:44   Log-Likelihood:                -36.243
No. Observations:                  23   AIC:                             76.49
Df Residuals:                      21   BIC:                             78.76
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      5.7052      1.615      3.532      0.0